sequencer/src/persistence/fs.rs

```diff
@@ -894,6 +901,42 @@ impl SequencerPersistence for Persistence {
     }
 }
 
+#[async_trait]
+impl DhtPersistentStorage for Persistence {
+    /// Save the DHT to the file on disk
+    ///
+    /// # Errors
+    /// - If we fail to serialize the records
+    /// - If we fail to write the serialized records to the file
+    async fn save(&self, records: Vec<SerializableRecord>) -> anyhow::Result<()> {
+        // Bincode-serialize the records
+        let to_save =
+            bincode::serialize(&records).with_context(|| "Failed to serialize records")?;
+
+        // Write the serialized records to the file
+        std::fs::write(self.inner.read().await.libp2p_dht_path(), to_save)
```

Comment:
This would fail if directory does not exist. We can use fs::create_dir_all() to create it first.
Also, we are using inner.replace() everywhere to wwrite the files